### PR TITLE
Add enter key support for `Find&Replace`

### LINKED ISSUE #5022: Find and Replace does not respond to the `Enter` key
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Open Find and Replace dialog.
2. Type something to find input.
3. Press `Enter` key.

### Expected result

After `Enter` key was pressed, the find should be invoked.

### Actual result

Nothing happens. It seems that only this dialog does not respond to the enter key because others like e.g. `Link` work properly.

## Other details

* Browser: Any
* OS: Any
* CKEditor version: ...
* Installed CKEditor plugins: Find

### PR DESCRIPTION
**Pull request by KarolDawidziuk**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

```
* [#5022](https://github.com/ckeditor/ckeditor4/issues/5022): Fixed: [Find/Replace](https://ckeditor.com/cke4/addon/find) Find and replace does not respond to the `Enter` key.
```

## What changes did you make?

Enter key not triggered find or replace because dialogs by default contain [handling only to dialog `OK` button](https://github.com/ckeditor/ckeditor4/blob/t/5022/plugins/dialog/plugin.js#L545-L554). To fix this I added `onKeyEvent` to the dialog inputs with executing `find` or `replace` method on current finder.

## Which issues does your PR resolve?

Closes #5022.

### CODE REVIEW

File: plugins/find/dialogs/find.js [hunk 1]
```diff
@@ -557,6 +557,61 @@
 		}
 
 		var lang = editor.lang.find;
+		function execFindOrReplace( dialog, type ) {
```
Comment (jacekbogdanski) [inline plugins/find/dialogs/find.js:560]: I would split find and replace to different functions for readability and have dedicated `execFind` / `exedReplace` functions.

File: plugins/find/dialogs/find.js [hunk 2]
```diff
@@ -678,7 +737,17 @@
 						label: lang.replaceWith,
 						isChanged: false,
 						labelLayout: 'horizontal',
-						accessKey: 'R'
+						accessKey: 'R',
+						onKeyDown: function( evt ) {
+							var keystroke = evt.data.getKeystroke();
+
+							if ( keystroke !== 13 ) {
```
Comment (jacekbogdanski) [inline plugins/find/dialogs/find.js:744]: It would be good to avoid magic numbers and extract `13` to constant (like you did in the unit test).

**Review by jacekbogdanski — CHANGES_REQUESTED** [review]
There is an issue with a replace handler as the match previously found by find is dismissed introducing inconsistency. E.g.

1. Find "bar" 2x.
2. Replace "bar" with "foo".

**Expected** The second "bar" occurrence found by the find method is replaced.
**Actual** Searching is restarted and the first "bar" occurrence is replaced.

https://user-images.githubusercontent.com/10279586/177781928-f5d71247-bc1b-4217-ba47-f254af51a76c.mov

**Comment by KarolDawidziuk** [comment]
Thanks, @jacekbogdanski for the review. 

I applied improvements from your comments and PR is ready for another check.

Unfortunately, for some reason, I was not able to reproduce the issue from the video. 🤔 Tried with a different approach.

**Review by jacekbogdanski — APPROVED** [review]

**Pull request merged**